src/main/scala/com/exasol/spark/util/Filters.scala

```diff
@@ -10,99 +14,85 @@ import org.apache.spark.sql.types._
 object Filters {
 
   /**
-   * Creates an Exasol SQL where clause from given list of
-   * [[org.apache.spark.sql.sources.Filter]]-s. Then these set of predicates
-   * will be pushed to Exasol for evaluation.
+   * Converts a sequence of filters into an Exasol boolean expressions.
    *
-   * @param schema A user provided or inferred schema of the query
-   * @param filters A sequence of Spark source filters
-   * @return A created Exasol where clause
+   * @param filters a sequence of Spark source filters
+   * @return a sequence of Exasol boolean expressions
    */
-  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
-  def createWhereClause(schema: StructType, filters: Seq[Filter]): String = {
-    val dataTypes = schema.map(field => field.name -> field.dataType).toMap
-    filters.flatMap(filterExpr(_, dataTypes)).mkString(" AND ")
-  }
+  def booleanExpressionFromFilters(filters: Seq[Filter]): Seq[BooleanExpression] =
+    filters.map(filterToBooleanExpression(_)).map(_.toList).flatten
 
   /**
-   * Given a Spark source [[org.apache.spark.sql.sources.Filter]], create a
-   * Exasol SQL expression.
+   * Given a Spark source [[org.apache.spark.sql.sources.Filter]],
+   * creates an Exasol boolean expression.
    *
-   * Returns [[scala.None]] if an expression could not be created from the
-   * filter.
+   * @param filter a Spark source filter
+   * @return an Exasol boolean expression, [[scala.None]] is returned if
```

Comment:
is `scala.None` == `null`?

Reply:
No. It is the "undefined" part of the Optional, e.g, this is empty `Optional` from Java.

The option class has to subtypes, `Some[T]` (usually used as `Option[T]`) and `None`, `Option(null)` converted to `None` type.

Reply:
I see